File: domain/src/main/kotlin/com/elorating/domain/league/League.kt
package com.elorating.domain.league

class League(var name: String) {
    var id: String? = null

    var settings: Settings = Settings()
        private set

    constructor(id: String, name: String) : this(name) {
        this.id = id
    }

    fun setAllowDraws(value: Boolean) {
        settings.allowDraws = value
    }

    fun setMaxScore(value: Int) {
        settings.maxScore = value
    }
}

File: webapp/src/app/shared/model/league.ts
import { Player } from './player';
import { Page } from './page';
import { CompletedGame, ScheduledGame } from './game';

export interface League {
  id: string;
  name: string;
  allowDraws: boolean;
  maxScore: number;
  players: Player[];
  completedGames: Page<CompletedGame>;
  scheduledGames: Page<ScheduledGame>;
}
